packages/circle-demo-webapp/app/lib/utils.ts

```diff
@@ -22,6 +23,27 @@ export async function callFetch<ReturnType extends object>(
     },
     body: JSON.stringify(body),
   });
+  if (res.status !== 200) {
+    const errorData = (await res.json()) as ErrorResponse;
+    throw new Error(errorData.error);
+  }
+  return (await res.json()) as ReturnType;
+}
+
+export async function callGetFetch<ReturnType extends object>(
```

Comment:
This isn’t consistent with the other hooks in our codebase. We need to standardize on a single format for the entire app. If you prefer using callGetFetch, please update the other hooks to match as well.

Reply:
yes. I think writing 20 lines of the same code for each request is not a good codebase. That's why I suggest to move  to callGetFetch and callFetch helpers

Reply:
also for useTransactions we have absolutely different solution `import useSWR, { mutate } from 'swr';` - also should be replaced to callGetFetch

Reply:
If you can ensure everything works consistently, with proper loading and error states, and prevent double-loading of the same data, then sure, go for it.

Reply:
fixed